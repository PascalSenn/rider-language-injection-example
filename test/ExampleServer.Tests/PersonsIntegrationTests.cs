using System.Threading.Tasks;
using HotChocolate;
using HotChocolate.Execution;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;
using Xunit;
using static ExampleServer.Tests.TestHelper;

namespace ExampleServer.Tests
{
    public class PersonsIntegrationTests
    {
        [Fact]
        public async Task GetPersons_Should_ReturnPagesPersons()
        {
            // arrange
            IServiceCollection services = ConfigureTestServices();
            IRequestExecutor executor = await services
                .AddGraphQLServer()
                .BuildRequestExecutorAsync();

            string query =
                @"query getPersons {
                persons {
                    nodes {
                        name
                    }
                }
            }";

            IReadOnlyQueryRequest request =
                QueryRequestBuilder.New().SetQuery(query).Create();

            // act
            IExecutionResult result = await executor.ExecuteAsync(request);

            // assert
            result.ToJson().MatchSnapshot();
        }
    }
}