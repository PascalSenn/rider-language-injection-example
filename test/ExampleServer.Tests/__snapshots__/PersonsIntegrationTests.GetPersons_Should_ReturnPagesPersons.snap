{
  "data": {
    "persons": {
      "nodes": [
        {
          "name": "Luke Skywalker"
        },
        {
          "name": "Darth Vader"
        }
      ]
    }
  }
}
